packages/tools/src/tools/ReferenceCursors.ts

```diff
@@ -0,0 +1,572 @@
+      );
+      renderStatus = true;
+    }
+
+    return renderStatus;
+  };
+
+  updateStackPosition(
+    viewport: Types.IStackViewport | Types.IVolumeViewport
+  ): void {
+    const currentMousePosition = this._currentCursorWorldPosition;
+
+    if (!currentMousePosition || currentMousePosition.some((e) => isNaN(e)))
+      return;
+    if (viewport instanceof StackViewport) {
+      const closestState = calculateMinimalDistanceForStackViewport(
+        currentMousePosition,
+        viewport
+      );
+
+      if (!closestState) return;
+      const { index } = closestState;
+      if (index !== viewport.getCurrentImageIdIndex())
+        viewport.setImageIdIndex(index);
+    } else if (viewport instanceof VolumeViewport) {
+      const { focalPoint, viewPlaneNormal } = viewport.getCamera();
+      if (!focalPoint || !viewPlaneNormal) return;
+      const plane = utilities.planar.planeEquation(viewPlaneNormal, focalPoint);
+      const currentDistance = planeDistanceToPoint(
+        plane,
+        currentMousePosition,
+        true
+      );
+
+      if (Math.abs(currentDistance) < 0.5) return;
+      const normalizedViewPlane = vec3.normalize(
+        vec3.create(),
+        vec3.fromValues(...viewPlaneNormal)
+      );
+      const scaledPlaneNormal = vec3.scale(
+        vec3.create(),
+        normalizedViewPlane,
+        currentDistance
+      );
+      const newFocalPoint = vec3.add(
+        vec3.create(),
+        vec3.fromValues(...focalPoint),
+        scaledPlaneNormal
+      ) as Types.Point3;
+      //TODO: make check if new focal point is within bounds of volume
+      const isInBounds = true;
+      if (isInBounds) {
+        viewport.setCamera({ focalPoint: newFocalPoint });
+        const renderingeEngine = viewport.getRenderingEngine();
+        if (renderingeEngine) renderingeEngine.renderViewport(viewport.id);
+      }
+    }
+  }
+}
+
+ReferenceCursors.toolName = 'ReferenceCursors';
+export default ReferenceCursors;
+
+//assumes that imageIds are sorted by slice location
+function calculateMinimalDistanceForStackViewport(
+  point: Types.Point3,
+  viewport: Types.IStackViewport
+): { distance: number; index: number } | null {
+  const imageIds = viewport.getImageIds();
+  const currentImageIdIndex = viewport.getCurrentImageIdIndex();
+
+  if (imageIds.length === 0) return null;
+
+  const getDistance = (imageId: string): null | number => {
+    const planeMetadata = getPlaneMetadata(imageId);
+    if (!planeMetadata) return null;
+    const plane = utilities.planar.planeEquation(
+      planeMetadata.planeNormal,
+      planeMetadata.imagePositionPatient
+    );
+    const distance = planeDistanceToPoint(plane, point);
+    return distance;
+  };
+
+  const closestStack = {
+    distance: getDistance(imageIds[currentImageIdIndex]) ?? Infinity,
+    index: currentImageIdIndex,
+  };
+
+  //check higher indices
+  const higherImageIds = imageIds.slice(currentImageIdIndex + 1);
```

Comment:
why are we splitting the stack into lower and upper sections? shouldn't we just do the search on all imageIds here? Also I believe this function is a useful function for other purposes (stack synchronizer tool), can we refactor it into the utils inside `@core/utilities/getClosestImageForPoint` or similar?

Also plane distance to point can also be pushed inside `@core/utilities/planar`